src/TPPCore.ChatProviders/Providers/Twitch/TwitchProvider.cs

```diff
@@ -85,16 +85,58 @@ public string GetUsername()
             var viewers = jsonDoc.SelectToken("chatters.viewers")
                 .Select(item => (string) item);
 
-            foreach (var usernames in new[] {moderators, staff, admins, global_mods, viewers})
+            PostRoomList roomList;
+
+            List<ChatUser> Viewers = new List<ChatUser> { };
+            List<ChatUser> Moderators = new List<ChatUser> { };
+            List<ChatUser> GlobalMods = new List<ChatUser> { };
+            List<ChatUser> Admins = new List<ChatUser> { };
+            List<ChatUser> Staff = new List<ChatUser> { };
+
+            foreach (string viewer in viewers)
+            {
+                var user = new ChatUser() { Username = viewer };
+                Viewers.Add(user);
+                users.Add(user);
+            }
+            foreach (string moderator in moderators)
+            {
+                var user = new ChatUser() { Username = moderator };
+                Moderators.Add(user);
+                users.Add(user);
+            }
+            foreach (string globalmoderator in global_mods)
             {
-                foreach (var username in usernames)
-                {
-                    var user = new ChatUser() { Username = username };
-                    users.Add(user);
-                }
+                var user = new ChatUser() { Username = globalmoderator };
+                GlobalMods.Add(user);
+                users.Add(user);
             }
+            foreach (string admin in admins)
+            {
+                var user = new ChatUser() { Username = admin };
+                Admins.Add(user);
+                users.Add(user);
+            }
+            foreach (string staffmember in staff)
+            {
+                var user = new ChatUser() { Username = staffmember };
+                Staff.Add(user);
+                users.Add(user);
+            }
+
+            var processedUsers = users.Distinct(new ChatUserEqualityComparer()).ToList();
```

Comment:
I think there might be some confusion about what the code is supposed to be doing here. 

Twitch IRC's room events are delayed or can be unavailable. What the code was doing was merging the IRC room list with a fresh call to the Twitch API. Because there is going to be duplicates, the Distinct method was supposed to make the room list contain unique `ChatUser`s. With the new changes, there isn't any deduplication besides the number of users in the room.